src/main/java/net/minecraftforge/common/ForgeModContainer.java

```diff
@@ -293,6 +295,20 @@ private static void syncConfig(boolean load)
         replaceVanillaBucketModel = prop.getBoolean(Boolean.FALSE);
         propOrder.add(prop.getName());
 
+        propOrder = new ArrayList<String>();
```

Comment:
Configs are generally poorly tested and supported, so it just adds a bit of complexity to the code without a guarantee of actually working. 
If a mod adds some really obnoxious effects, people can complain to them for a config. I doubt many people would look into the Forge configs.